client/src/containers/ProtectedRoute/ProtectedRoute.js

```diff
@@ -50,6 +49,11 @@ class ProtectedRoute extends React.Component {
         else if (oneOf && !oneOf.includes(session.user.role)) {
             render = () => noAccessRedirect('You do not have sufficient privileges to view that page.');
         }
+        // If the user has not verified their email and email verification is required for the route,
+        // redirect them
+        else if (verifyEmail && session.user && !session.user.isEmailVerified) {
+            render = () => noAccessRedirect('You must verify your email address to view that page.');
```

Comment:
We could link to https://monuments.us.org/account or https://monuments.us.org/signup/confirm?resend=true so that they can resend the verification email